WireAPI/Sources/WireAPI/APIs/FeatureConfigsAPI/FeatureConfigsAPIV0.swift

```diff
@@ -0,0 +1,164 @@
+
+    }
+
+}
+
+struct FeatureConfigsResponseAPIV0: Decodable, ToAPIModelConvertible {
+
+    let appLock: FeatureWithConfig<FeatureConfigResponse.AppLockV0>
+    let classifiedDomains: FeatureWithConfig<FeatureConfigResponse.ClassifiedDomainsV0>
+    let conferenceCalling: FeatureWithoutConfig
+    let conversationGuestLinks: FeatureWithoutConfig
+    let digitalSignatures: FeatureWithoutConfig
+    let fileSharing: FeatureWithoutConfig
+    let selfDeletingMessages: FeatureWithConfig<FeatureConfigResponse.SelfDeletingMessagesV0>
+    let mls: FeatureWithConfig<FeatureConfigResponse.MLSV0>?
+
+    func toAPIModel() -> [FeatureConfig] {
+        var featureConfigs: [FeatureConfig] = []
+
+        let appLockConfig = AppLockFeatureConfig(
+            status: appLock.status,
+            isMandatory: appLock.config.enforceAppLock,
+            inactivityTimeoutInSeconds: appLock.config.inactivityTimeoutSecs
+        )
+
+        featureConfigs.append(.appLock(appLockConfig))
+
+        let classifiedDomainsConfig = ClassifiedDomainsFeatureConfig(
+            status: classifiedDomains.status,
+            domains: classifiedDomains.config.domains
+        )
+
+        featureConfigs.append(.classifiedDomains(classifiedDomainsConfig))
+
+        let conferenceCallingConfig = ConferenceCallingFeatureConfig(
+            status: conferenceCalling.status,
+            useSFTForOneToOneCalls: nil
+        )
+
+        featureConfigs.append(.conferenceCalling(conferenceCallingConfig))
+
+        let conversationGuestLinksConfig = ConversationGuestLinksFeatureConfig(
+            status: conversationGuestLinks.status
+        )
+
+        featureConfigs.append(.conversationGuestLinks(conversationGuestLinksConfig))
+
+        let digitalSignaturesConfig = DigitalSignatureFeatureConfig(
+            status: digitalSignatures.status
+        )
+
+        featureConfigs.append(.digitalSignature(digitalSignaturesConfig))
+
+        let fileSharingConfig = FileSharingFeatureConfig(
+            status: fileSharing.status
+        )
+
+        featureConfigs.append(.fileSharing(fileSharingConfig))
+
+        let selfDeletingMessagesConfig = SelfDeletingMessagesFeatureConfig(
+            status: selfDeletingMessages.status,
+            enforcedTimeoutSeconds: selfDeletingMessages.config.enforcedTimeoutSeconds
+        )
+
+        featureConfigs.append(.selfDeletingMessages(selfDeletingMessagesConfig))
+
+        if let mls {
+            let mlsConfig = MLSFeatureConfig(
+                status: mls.status,
+                protocolToggleUsers: mls.config.protocolToggleUsers,
+                defaultProtocol: mls.config.defaultProtocol,
+                allowedCipherSuites: mls.config.allowedCipherSuites,
+                defaultCipherSuite: mls.config.defaultCipherSuite,
+                supportedProtocols: [.proteus] /// Default to Proteus
+            )
+
+            featureConfigs.append(.mls(mlsConfig))
+        }
```

Comment:
I'm pretty sure mls isn't in v0. Is Android also parsing the mls config in v0?

Reply:
while this property is optional on Android, the same `FeatureConfigResponse` is parsed across different api versions however provides a default value when there's no value. Swagger doesn't give us any information on that matter. Should I remove it from v0 or align with Android ?

`    override fun fromDTO(data: FeatureConfigData.MLS?): MLSModel =
        data?.let {
            MLSModel(
                defaultProtocol = it.config.defaultProtocol.toModel(),
                supportedProtocols = it.config.supportedProtocols.map { it.toModel() }.toSet(),
                status = fromDTO(it.status),
                supportedCipherSuite = SupportedCipherSuite(
                    default = CipherSuite.fromTag(it.config.defaultCipherSuite),
                    supported = it.config.allowedCipherSuites.map { CipherSuite.fromTag(it) }
                )
            )
        } ?: MLSModel(
            defaultProtocol = SupportedProtocol.PROTEUS,
            supportedCipherSuite = null,
            supportedProtocols = setOf(SupportedProtocol.PROTEUS),
            status = Status.DISABLED
        )`

Reply:
So given backend team feedback, I did some changes. MLS starts at v1, MLS is no longer optional.
MLS migration and E2EI stars at v4, they're no longer optionals. All top level objects are non optional but stars at different api versions (e.g v1, v4).

Reply:
Yeah I think that's the way to go, that way we don't risk to carry optional types into the future when they're no longer needed.